Update/v15 - New Timelines chapter (#295)
AvolitesLtd/TitanManual: docs/timelines/running-and-editing-timelines.md

---
id: running-and-editing-timelines
title: Running and Editing Timelines
sidebar_label: Running and Editing Timelines
---

Running a timeline to timecode
------------------

The HTP (intensity) of fixtures in the timeline is controlled by the fader level of the handle. 

1.  Ensure the **timecode link** button (bottom left of the timeline view) is set to **Linked**.

> 2.  Push up the fader of the handle where the timeline is stored.

Reviewer: This step is no longer required if you are using the "Activate in range" option which is enabled by default.

3.  Start the timecode source. 

4.  The cues in the timeline will run when the timecode reaches their programmed times.

- On consoles with motorised faders and on virtual faders, the faders will move as programmed.

- If you are using internal timecode you can set the timecode to run automatically when you raise the fader, and/or to stop when you lower the fader. See [Timeline Options - Fader](../timelines/timeline-options.md#fader-tab).

- You can set the timeline to loop (when using internal timecode), see [Timeline Options - Timecode](../timelines/timeline-options.md#timecode-tab).

> - If you start the timecode at a point part way through the timeline, any active triggers will fire, but previous triggers will not - this might result in fixtures starting from a different state and changing the look.

Reviewer: Might be worth explaining that this is the case because of LTP behaviour and mentioning that you can mostly get around it by using the release options

### Testing a timeline

You can test a timeline without running the timecode source by setting the **timecode link** button to **Unlinked**. You can then use the Play/Pause/Rewind buttons
in the Timeline View window to run the timeline sequence. This can be useful when you don't have control of the timecode source.                  

Timeline Editing
--------------------------
 
### Selecting triggers

To edit a trigger, you need to select it. There are several different ways to do this.

> -   Click on a trigger block to select all the trigger points within the block. You will see selection 

Reviewer: Click -> tap

Reviewer: Or maybe "press"

Author: Changed to "Tap" as that is used elsewhere. "Press" is used for touch buttons.

![Timeline trigger handles](/docs/images/Timeline-Handles.png)

- Click on more triggers to add them to the selection.

- Draw a selection marquee box over the trigger points you want to select. Only the trigger points enclosed by the selection box will be selected, so if you want to select all points in a block they all need to be inside the box. This is a quick way to select one trigger point without having to "multi-click" on a block.

- To clear all selection use the context menu button \{Clear Trigger Selection\} or click in an empty area of the Timeline View.

### Using arrow buttons to move trigger selection

The left/right arrow buttons for the track will move your selection to the previous trigger point or the next trigger point.

- Left arrow - moves selection to the previous trigger point on the track.
- Right arrow - moves selection to the next trigger point on the track.

If nothing is selected, the right arrow will select the first trigger on the track and the left arrow will select the last.

### Using the wheels to edit triggers

When you have one or more trigger points selected, the wheels control the time, level and fade of those triggers.

> - Left hand wheel sets the time.

Reviewer: This will vary depending on what type of trigger you have selected. I think you're right that this is the only option now though.

- Middle wheel sets the level.
- Right hand wheel sets the fade.

You can change multiple triggers at the same time using the wheels.

### Using the overview bar

The **Overview Bar** at the bottom of the Timeline View gives you a miniature view of all the events in the track.

![Timeline overview bar](/docs/images/Timeline-Overview-Bar.png)

- The handles at the left and right of the Overview Bar set the start and end points of the main view above. This gives you an easy way to zoom in on a particular area.

- When you are zoomed in you can drag on the zoomed area in the bar to move the main view (the same as using the Hand tool and dragging on the main view).

- The edit cursor is shown in grey.

- The current live time is shown in blue.

### Deleting triggers

> You can delete trigger blocks by pressing \<Delete\> then tapping on the block you want to delete.

Reviewer: Marquee select is also an option for deletion

You can also delete tracks or delete rows from tracks using this method.

### Setting track legends and halo colours on tracks

To help you identify different tracks, you can set a halo colour for each track. This colour outlines the track and also sets the colours of the trigger blocks which makes them easier to see in the overview bar.

![Timeline halo](/docs/images/Timeline-Halo.png)

1. Press \[Set Legend\] at the top level main menu.

2. Tap on the Track number in the Timeline View.

3. Change \[Legend\] to set the track legend or press \[Halo\].

4. Select a halo colour from the colour picker.

- You can remove a halo colour by pressing \[Remove Halo\] at step 4.

> - You can also set legends on markers using the \[Set Legend\] button.

Reviewer: You can set the halo of playbacks and wait for go triggers too.

### Snap options

Using the context menu option \{Snap Options\} you can configure whether new triggers will
snap to existing objects when you enter them by clicking on the Timeline View.
The snap options are
- Snap To Triggers
- Snap To Markers
- Snap To Cursor.

### Collapsing track view

If you have a lot of tracks, you can collapse them to a single line to show more tracks on the screen, by clicking the arrow button to the left of the track name.

### Table view

You can switch the Timeline View to a table version using the \{Open Table View\} context button. 

> This shows details for each trigger in time order, in the same format as a cue list. Buttons down the left hand side allow you to filter which tracks are shown in the list.

Reviewer: Again less switch more open another view, you can have both at once.

---
id: running-and-editing-timelines
title: Running and Editing Timelines
sidebar_label: Running and Editing Timelines
---

Running a timeline to timecode
------------------

> The HTP (intensity) of fixtures in the timeline is controlled by the fader level of the handle. The console will automatically fire the timeline playback to full when timecode is received within the range of the timeline settings (you can turn this off using the option [Autoload in Range](../timelines/timeline-options.md#timecode-tab)).

Reviewer 2: It has now been renamed "Activate in range" (#87417) to avoid confusion with Cuelist autoloading

1.  Ensure the **timecode link** button (bottom left of the timeline view) is set to **Linked**.

> 2.  Start the timecode source. (You will need to push up the timeline fader for internal timecode or if you have disabled the autoload option described above).

Reviewer 2: Do we not auto-activate if the timecode source is internal? That sounds like it might be a bug?

Author: As far as I can see the internal timecode does not activate by default. You can set the Fader Raised Action option to Play to make this happen but it isn't the default setting. Tell me if I am missing somethinig.

Reviewer: When we say activate we're talking about not needing to push up the fader. i.e. when you start the timecode source the fader should automatically go to full. So this part is not needed: `(You will need to push up the timeline fader for internal timecode or if you have disabled the autoload option described above).`

Author: OK, should I leave in "You will need to push up the timeline fader if you have disabled the auto-activate option described above"

Reviewer: Yep, that part is correct 👍

> 3.  The cues in the timeline will run when the timecode reaches their programmed times.

Reviewer 2: ```suggestion
3.  The triggers in the timeline will fire when the timecode reaches their programmed times.
```

    handles appear.  Tap again on a selected trigger block to step through the
    trigger points within the block. 

![Timeline trigger handles](/docs/images/Timeline-Handles.png)

- Tap on more triggers to add them to the selection.

- Draw a selection marquee box over the trigger points you want to select. Only the trigger points enclosed by the selection box will be selected, so if you want to select all points in a block they all need to be inside the box. This is a quick way to select one trigger point without having to "multi-tap" on a block.

- To clear all selection use the context menu button \{Clear Trigger Selection\} or tap in an empty area of the Timeline View.

### Using arrow buttons to move trigger selection

The left/right arrow buttons for the track will move your selection to the previous trigger point or the next trigger point.

- Left arrow - moves selection to the previous trigger point on the track.
- Right arrow - moves selection to the next trigger point on the track.

If nothing is selected, the right arrow will select the first trigger on the track and the left arrow will select the last.

### Using the wheels to edit triggers

> When you have one or more trigger points selected, the wheels control the time, level and fade of those triggers (some types of trigger do not have all these parameters).

Reviewer 2: ```suggestion
When you have one or more trigger points selected, the wheels control the timecode time, target level and fade time of those triggers (some types of trigger do not have all these parameters).
```

> - Wheel A sets the time.
> - Wheel B sets the level.
> - Wheel C sets the fade.

Reviewer 2: ```suggestion
- Wheel A sets the timecode time.
- Wheel B sets the target level.
- Wheel C sets the fade time.
```

You can change multiple triggers at the same time by selecting them all and using the wheels.

> ### Using the overview bar

Reviewer 2: Is this not somewhat duplicated from `docs/timelines.md`? Might be good to consolidate

Author: The original section introduces the basics of the overview bar - this section has more detail. Personally I think it is worth having both sections for the skim reader and those who want more.

The **Overview Bar** at the bottom of the Timeline View gives you a miniature view of all the events in the track.

![Timeline overview bar](/docs/images/Timeline-Overview-Bar.png)

- The handles at the left and right of the Overview Bar set the start and end points of the main view above. This gives you an easy way to zoom in on a particular area.

- When you are zoomed in you can drag on the zoomed area in the bar to move the main view (the same as using the Hand tool and dragging on the main view).

- The edit cursor is shown in grey.

- The current live time is shown in blue.

### Deleting triggers

You can delete trigger blocks by pressing \<Delete\> then tapping on the block you want to delete, or drawing a marquee selection box over the triggers to be deleted.

You can also delete tracks or delete rows from tracks using this method.

### Setting track legends and halo colours on tracks

To help you identify different tracks, you can set halo colours. This colour outlines the track and also sets the colours of the trigger blocks which makes them easier to see in the overview bar.

> In addition you can set different colours for the playback actions within each track as shown in Track 2 in the picture below.

Reviewer 2: I'm not sure what `playback actions` refers to here... I don't think `action` is a term that we've used elsewhere?

Author: I mean the row in the track that refers to a particular playback. Would it be better to say "you can set different colours for each playback row within the track"?

Reviewer 2: Ah, I see what you mean. What's actually happening here is that you're setting the halo of the playback itself - you can test this by setting the halo colour of a playback and then assigning it into the timeline. The playback's halo colour gets shown in the timeline.

Author: Right, didn't spot that, will update

![Timeline halo](/docs/images/Timeline-Halo.png)

1. Press \[Set Legend\] at the top level main menu.

2. Tap on the Track legend or the playback row in the Timeline View.

3. Change \[Legend\] to set the track legend or press \[Halo\].

4. Select a halo colour from the colour picker.

- You can remove a halo colour by pressing \[Remove Halo\] at step 4.

- You can also set legends on markers using the \[Set Legend\] button.

- You can set a halo colour for a **Wait for Go** trigger by tapping on it at step 2.

### Snap options

Using the context menu option \{Snap Options\} you can configure whether new triggers will
snap to existing objects when you enter them by clicking on the Timeline View.

> The snap options are

Reviewer 2: ```suggestion
The snap options are:
```

- Snap To Triggers
- Snap To Markers

> - Snap To Cursor.

Reviewer 2: ```suggestion
- Snap To Cursor
```

### Collapsing track view

> If you have a lot of tracks, you can collapse them to a single line to show more tracks on the screen, by clicking the arrow button to the left of the track name.

Reviewer 2: Might be good to describe what the semi-collapsed state does.

Perhaps deduplicate from `docs/timeline.md`?

Author: Semi-collapsed was not in the version I was testing... I see it now, what does it actually do? It seems to sometimes combine playback rows in the track and sometimes not but I can't work out when each happens.

Reviewer 2: So it tries to lay out the playback level groups in the track using the fewest number of rows possible; so
```
[---A--]
      [---B---]
             [---C---]
```
would instead be laid out
```
[---A--] [---C---]
      [---B---]
```

In the uncollapsed state at the top, each playback has its own row, in the collapsed state at the bottom they share as can be made to fit.
